### PR TITLE
fix #1465: Difference of structure between ListIssuesOptions, ListGro…

### LINKED ISSUE #1465: Difference of structure between ListIssuesOptions, ListGroupIssuesOptions, and ListProjectIssuesOptions
There is a type difference for `NotMilestone` item in `ListIssuesOptions`, `ListGroupIssuesOptions`, and `ListProjectIssuesOptions` structure: `*string` or `*[]string`.

From `issues.go` file:
```go
type ListIssuesOptions struct {
	ListOptions
       ...
	NotMilestone     *string `url:"not[milestone],omitempty" json:"not[milestone],omitempty"`
       ...
}

type ListGroupIssuesOptions struct {
	ListOptions
       ...
	NotMilestone     *string `url:"not[milestone],omitempty" json:"not[milestone],omitempty"`
       ...
}

type ListProjectIssuesOptions struct {
	ListOptions
       ...
	NotMilestone       *[]string        `url:"not[milestone],omitempty" json:"not[milestone],omitempty"`
       ...
}
```

I think it should be the same.

### PR DESCRIPTION
**Pull request by cyd01**
fix #1465: Difference of structure between ListIssuesOptions, ListGroupIssuesOptions, and ListProjectIssuesOptions

### CODE REVIEW

File: issues.go [hunk 1]
```diff
@@ -322,7 +322,7 @@ type ListProjectIssuesOptions struct {
 	NotLabels          *Labels          `url:"not[labels],comma,omitempty" json:"not[labels],omitempty"`
 	WithLabelDetails   *bool            `url:"with_labels_details,omitempty" json:"with_labels_details,omitempty"`
 	Milestone          *string          `url:"milestone,omitempty" json:"milestone,omitempty"`
-	NotMilestone       *[]string        `url:"not[milestone],omitempty" json:"not[milestone],omitempty"`
+	NotMilestone       *string          `url:"not[milestone],omitempty" json:"not[milestone],omitempty"`
```
Comment (svanharmelen) [inline issues.go:325]: I think you should correct the other ones as this one was the correct one...

Comment (cyd01) [reply issues.go:325]: I finally think you are right: all (Labels, Milestones ...) seem to be slices. But I've never use it like that. I'll do some more tests.  
Feel free to reject that PR if you wish. After my tests, I'll be back with another one.

Comment (cyd01) [reply issues.go:325]: I've done some more tests. `NotLabels` and others ones except `NotMilestone` accept slice of strings.

    /api/v4/projects/100/issues?not[labels]=label1,label3

Since an issue can only be related to one milestone, `Milestone` in ListXXXIssuesOptions structure is definitely a (simple) string.  But `NotMilestone` could have been a slice of string. Unfortunately it does not work. The API seems to accept a string only.

    /api/v4/projects/100/issues?not[milestone]=milestone2

In conclusion you are allowed to give exactly one milestone in `NotMilestone` field.

I think my Pull request is stile valid.

Comment (svanharmelen) [reply issues.go:325]: Check... Thanks for going the extra mile on this one!

**Review by svanharmelen — APPROVED** [review]
